src/functionals.jl

```diff
@@ -32,3 +32,12 @@ end
 function kldivergence(P::UnivariateDistribution, Q::UnivariateDistribution)
     expectation(P, x -> let p = pdf(P,x); (p > 0)*log(p/pdf(Q,x)) end)
 end
+
+function kldivergence(P::MultivariateDistribution, Q::MultivariateDistribution; n_samples=100)
+    mcexpectation(P, x -> let p = pdf(P,x); (p > 0) * log(p) - logpdf(Q, x) end; n_samples)
+end
+
+function mcexpectation(distr::Distribution, f; n_samples::Int=100)
+    n_samples > 0 || throw(ArgumentError("number of samples should be > 0"))
+    sum(f, rand(distr) for _ in 1:n_samples) / n_samples
+end
```

Comment:
I think `mean(f, itr)` only exists since 1.3

Reply:
Also what you wrote (`xlogx((p > 0) * log(p))`) is wrong, since we sample from `p`, `p(x)` should not be part of the function

Reply:
> I think mean(f, itr) only exists since 1.3

Ah yes, I missed this note in the docstring :+1:

> Also what you wrote (xlogx((p > 0) * log(p))) is wrong, since we sample from p, p(x) should not be part of the function

Ah, yes, I didn't mean to, I guess I copied something incorrectly.

The better suggestion:
```suggestion
function kldivergence(P::MultivariateDistribution, Q::MultivariateDistribution; kwargs...)
    expectation(P, x -> let p = pdf(P,x); (p > 0) * (log(p) - logpdf(Q, x)) end; kwargs..)
end

# Monte Carlo fallback
function expectation(distr::MultivariateDistribution, f; nsamples::Int=100, rng::AbstractRNG=GLOBAL_RNG)
    nsamples > 0 || throw(ArgumentError("number of samples should be > 0"))
    return _expectation(rng, f, sampler(distr), nsamples) 
end

# Function barrier for `sampler` type instabilities
_expectation(rng, f, sampler, n) =  sum(f, rand(rng, sampler) for _ in 1:n) / n
```